From 5ed34e7090dfe69ebac6366efb2dbc3006f42831 Mon Sep 17 00:00:00 2001
From: Rosen Penev <rosenp@gmail.com>
Date: Tue, 17 Sep 2024 16:59:28 -0700
Subject: [PATCH] arch: arm: kirkwood: add nvmem mac address

of_get_ethdev_address gets called too early for nvmem. Allow the ethernet
driver to fix this.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
---
 arch/arm/mach-mvebu/kirkwood.c | 76 ----------------------------------
 1 file changed, 76 deletions(-)

--- a/arch/arm/mach-mvebu/kirkwood.c
+++ b/arch/arm/mach-mvebu/kirkwood.c
@@ -87,13 +87,18 @@ static void __init kirkwood_dt_eth_fixup
 		void __iomem *io;
 		u8 *macaddr;
 		u32 reg;
+		int err;
 
 		if (!pnp)
 			continue;
 
-		/* skip disabled nodes or nodes with valid MAC address*/
-		if (!of_device_is_available(pnp) ||
-		    !of_get_mac_address(np, tmpmac))
+		/* skip disabled nodes*/
+		if (!of_device_is_available(pnp))
+			goto eth_fixup_skip;
+
+		/* skip nodes with valid MAC address*/
+		err = of_get_mac_address(np, tmpmac);
+		if (err == -EPROBE_DEFER || !err)
 			goto eth_fixup_skip;
 
 		clk = of_clk_get(pnp, 0);
